ui/app/utils/no-leader-error.js

```diff
@@ -0,0 +1,11 @@
+import { AdapterError } from 'ember-data/adapters/errors';
+
+export const NO_LEADER = 'No cluster leader';
+
+const NoLeaderError = function() {
+  AdapterError.call(this, [], NO_LEADER);
+};
+
+NoLeaderError.prototype = Object.create(AdapterError.prototype);
```

Comment:
I'm surprised to see this implementation. Is this new public API for extending `DS.AdapterError`? I expected to see `export default DS.AdapterError.extend({ … })` here…

Reply:
I ripped it [from source](https://github.com/emberjs/data/blob/master/addon/-private/adapters/errors.js#L95-L104), but I agree that it's unusual.

Not sure why I didn't consider plain ol' `.extend`. I guess I didn't imagine `AdapterError` as an Ember Object? I'll try it out now.

Reply:
Success! Thanks for the tip.